Refactor request

weronika-kurczyna opened the pull request:
Done: 
- TODO notes 
- added checking if the campaign exists (Reward Promotion) in main server.js
- added showing an alert when user doesn't have the specific campaign after adding new product (tiered-cart-promotions) which allows user to proceed to our github docs where the problem is explained
<img width="1056" alt="image" src="https://user-images.githubusercontent.com/117282008/201942660-8eb71297-3349-46f7-887a-1d908d3f3aca.png">

Commit:
refactor requests

weronika-kurczyna committed:
Change "attachEndpointsStackingPromotions" to "addStackingPromotionRoutes", remove "accessToStackingPromotionApp". Change names of "promotionStackableObj" and "defaultItems". Move "validateStackableParams" and "customer" objects to "addStackingPromotionRoutes" function.

weronika-kurczyna committed:
Add validateStackableParams and redeemStackableParams

Add validateStackableParams and redeemStackableParams instead of one global object

weronika-kurczyna committed:
Add check if campaign 'Reward Promotion' exists

weronika-kurczyna committed:
Add alert with information about lack of the campaign

marcin-slezak commented on server.js:
```diff
@@ -41,16 +41,30 @@ const checkCredentials = async () => {
+        await client.campaigns.get("Reward Promotion");
+    } catch (error) {
+        if (error.code === 404) {
+            const msg = "The 'Reward Promotion' campaign not found. This campaign is required for 'tiered-cart-promotion' to work properly. Please create a 'Reward Promotion' campaign first, you can check the details by visiting `https://docs.voucherify.io/reference/create-campaign`";
```
Just wondering, can you give a hint about the `Reward Promotion` configuration? Maybe we can put an example `curl` that can create this promotion as a comment here so developers facing this issue can create it quickly?

marcin-slezak commented on stacking-promotions/server.js:
```diff
@@ -42,17 +33,26 @@ export const attachEndpointsStackingPromotions = (app, client) => {
         const products = req.body.items;
         const items = mapInputIntoKnownProducts(products);
+
+        const validateStackableParams = {
```
we do not have to split preparing `validateStackableParams` into two steps (defining structure + update values). We can do that in one step.

```ts
const validateStackableParams = {
  order: {
    amount: calculateCartTotalAmount(items)
  },
  redeemables: removeDuplicatedPromoObjects(redeemables),
  customer
}

```

the same for redeem endpoint

weronika-kurczyna committed:
Reduction of the code

Assigning params directly in object.

weronika-kurczyna committed:
add cURL

weronika-kurczyna committed:
reformat cURL

weronika-kurczyna committed:
Update README.md

weronika-kurczyna committed:
Add information in server.js

weronika-kurczyna committed:
Delete the alert message

I deleted the alert because it started generating bugs when checking if the rewardPromotion exists.

weronika-kurczyna committed:
Refactor other endpoints
